File: SchoolWebApp-Client/src/app/academics/models/exam-name.ts
import {ResourceModel} from '@/core/models/ResourceModel';
import { ExamType } from './exam-type';

export class ExamName extends ResourceModel<ExamName> {
    public name?: string;
    public examTypeId?: number;
    public examType?: ExamType;
    public rank?: number;
    public description?: string;

    constructor(model?: Partial<ExamName>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/Academics/ExamName.cs
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Academics
{
    public class ExamName: SettingsBase
    {
        public int ExamtypeId { get; set; }
        public ExamType? ExamType { get; set; }

        public List<Exam> Exams { get; set; } = new();
    }
}
